README.md

```diff
@@ -85,24 +101,105 @@ let client = OpenFeatureAPI.shared.getClient()
+
+// add a hook for this evaluation only
+        _ = client.getValue(
+            key: "key",
+            defaultValue: false,
+            options: FlagEvaluationOptions(hooks: [ExampleHook()]))
+```
 
-### Events
+### Eventing
 
 Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
-Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are emitted for every provider.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
 Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
-Please refer to the documentation of the provider you're using to see what events are supported and when they are emitted.
 
-To register a handler for API level provider events, use the `OpenFeatureAPI` `addHandler(observer:selector:event:)` function. Event handlers can also be removed using the equivalent `removeHandler` function. `Client`s also provide add and remove functions for listening to that specific client. A `ProviderEvent` enum is defined to ease subscription.
+Please refer to the documentation of the provider you're using to see what events are supported.
 
-Events can contain extra information in the `userInfo` dictionary of the notification. All events will contain a reference to the provider that emitted the event (under the `providerEventDetailsKeyProvider` key). Error events will also provide a reference to the underlying error (under the `providerEventDetailsKeyError` key). Finally, client specific events will contain a reference to the client (under the `providerEventDetailsKeyClient` key)
+```swift
+OpenFeatureAPI.shared.addHandler(
+    observer: self, selector: #selector(readyEventEmitted(notification:)), event: .ready
+)
 
-### Providers
+func readyEventEmitted(notification: NSNotification) {
+    // to something now that the provider is ready
+}
+```
+
+### Shutdown
+
+The OpenFeature API provides a close function to perform a cleanup of all registered providers.
+This should only be called when your application is in the process of shutting down.
 
-To develop a provider, you need to create a new project and include the OpenFeature SDK as a dependency. This can be a new repository or included in the existing contrib repository available under the OpenFeature organization. Finally, you’ll then need to write the provider itself. This can be accomplished by implementing the `FeatureProvider` protocol exported by the OpenFeature SDK.
+<!-- TODO: code example for global shutdown -->
+
+## Extending
+
+### Develop a provider
+
+To develop a provider, you need to create a new project and include the OpenFeature SDK as a dependency.
+This can be a new repository or included in [the existing contrib repository](https://github.com/open-feature/swift-sdk-contrib) available under the OpenFeature organization.
```

Comment:
note: Contrib repo has not been created yet

Reply:
I would omit this whole sentence for now, then. We can always add it back in if/when we have a contribs repo.